src/rust/engine/src/context.rs

```diff
@@ -94,8 +96,8 @@ impl Core {
         )),
       };
 
-    // TODO: Allow configuration of process concurrency.
-    let command_runner = BoundedCommandRunner::new(underlying_command_runner, 16);
+    let command_runner =
+      BoundedCommandRunner::new(underlying_command_runner, process_execution_parallelism);
```

Comment:
Github won't let me comment on it, but:

Line 90 should change from `1` to `process_execution_parallelism + 2` or something along those lines (`+2` or similar for any book-keeping threads that grpc may need)